app/assets/javascripts/discourse/app/components/glimmer-site-header.gjs

```diff
@@ -452,6 +454,49 @@ export default class GlimmerSiteHeader extends Component {
   </template>
 }
 
+const INITIAL_DELAY_MS = 50;
+const DEBOUNCE_MS = 5;
+class Docking {
+  dockCheck = null;
+  _initialTimer = null;
+  _queuedTimer = null;
+
+  constructor(dockCheck) {
+    this.dockCheck = dockCheck;
+    window.addEventListener("scroll", this.queueDockCheck, { passive: true });
+
+    // dockCheck might happen too early on full page refresh
+    this._initialTimer = discourseLater(
+      this,
+      this._safeDockCheck,
+      INITIAL_DELAY_MS
+    );
+  }
+
+  @debounce(DEBOUNCE_MS)
+  queueDockCheck() {
+    this._queuedTimer = this._safeDockCheck;
+  }
+
+  @action
+  _safeDockCheck() {
+    if (this.isDestroyed || this.isDestroying) {
```

Comment:
This helper object doesn't have `isDestroyed` or `isDestroying` properties, so this will always be false.

It looks like we now have a `destroy()` method which cancels all the timers and event listeners. That should provide enough safety by itself, so I reckon we just remove this `_safeDockCheck` function and just call `dockCheck()` directly.